compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -403,6 +409,15 @@ public BallerinaTypeDefinitionSymbol createTypeDefinition(BTypeSymbol typeSymbol
             symbolBuilder.withQualifier(Qualifier.PUBLIC);
         }
 
+        if (typeSymbol instanceof BTypeDefinitionSymbol) {
```

Comment:
Is it not possible to check the `typeSymbol.kind == SymbolKind.TYPE_DEF` ?

Reply:
Did instanceOf to avoid null check before kind check. Will check and update, not null cases

Reply:
Updated in 5769c86